.github/workflows/build-gem.yml

```diff
@@ -0,0 +1,160 @@
+name: Build gem
+
+on:
+  workflow_dispatch:
+    inputs:
+      push:
+        description: Push gem
+        required: true
+        type: boolean
+        default: true
+  push:
+    branches:
+      - "**"
+
+env:
+  GEM_HOST: 'https://rubygems.pkg.github.com/DataDog'
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        type:
+          - final
+          - alpha
```

Comment:
Regarding this `alpha` name:

I used to find it fitting semantically: we do `beta` as part of a manual release process, so these versions are more `alpha` and thus should sort before? But I'm not so convinced anymore that such ordering matters, or is even consistent with reality.

I also found it lacking, because `1.42.0.alpha.foobar` kind of makes one think that `1.42.0` is the next version when it is actually the last one released.

That's why I also liked `dev` (because they emanate from unmerged branches) or `ci` (because they are built only from CI).

Examples to compare and contrast (taking `1.10.1` instead of the current `1.11.0.beta1` which muddies the thing):

- `1.10.1.alpha.gha4745620001.gfe6c0807f40b.foo.bar.branch.name`
- `1.10.1.ci.gha4745620001.gfe6c0807f40b.foo.bar.branch.name`
- `1.10.1.dev.gha4745620001.gfe6c0807f40b.foo.bar.branch.name`

So I'm increasingly inclined to move from `alpha` to either `dev` or `ci`. WDYT?

Reply:
@lloeki Thanks for the reasoning. 

I'm more inclined to `dev`. For me, it stipulates that this version is still under development and is not ready to be used in production and is just for development purposes

Reply:
Agreed. Done.